CRM/Utils/System/Joomla.php

```diff
@@ -784,8 +795,8 @@ public function getTimeZoneString() {
    */
   public function getModules() {
     $result = [];
-
-    $db = JFactory::getDbo();
+    $factoryClassName = $this->factoryClassName();
+    $db = $factoryClassName::getDbo();
```

Comment:
`Factory::getDbo()` is deprecated in Joomla 4.3. Suggest replacing with `Factory::getContainer()->get(DatabaseInterface::class)`.

https://api.joomla.org/cms-4/classes/Joomla-CMS-Factory.html#method_getDbo